CMakeLists.txt

```diff
@@ -126,10 +129,38 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 #
 # cmake ... -DCMAKE_INSTALL_PREFIX=$HOME/opt
 #
-include(GNUInstallDirs)
 if (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
   set(CMAKE_INSTALL_PREFIX "${CMAKE_BINARY_DIR}/install" CACHE PATH "..." FORCE)
 endif()
+include(GNUInstallDirs)
+
+
+# Configure how the SDK will manage runtime paths. This is important
+# even for BUILD_SHARED_LIBS=OFF due to the always-shared
+# `libviam_rust_utils` shared library.
+set(CMAKE_BUILD_WITH_INSTALL_RPATH false)
+set(CMAKE_BUILD_RPATH_USE_ORIGIN true)
+set(CMAKE_INSTALL_RPATH_USE_LINK_PATH true)
+set(CMAKE_MACOSX_RPATH true)
+set(CMAKE_SKIP_BUILD_RPATH false)
+
+list(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}" isSystemDir)
```

Comment:
This section then actually sets the correct install name and RPATH values based on the layout of the installation directory.